code/backend/batch/utilities/tools/QuestionAnswerTool.py

```diff
@@ -20,31 +30,115 @@ def __init__(self) -> None:
         self.name = "QuestionAnswer"
         self.vector_store = AzureSearchHelper().get_vector_store()
         self.verbose = True
+        self.env_helper = EnvHelper()
+        self.config = ConfigHelper.get_active_config_or_default()
 
-    def answer_question(self, question: str, chat_history: List[dict], **kwargs: dict):
-        config = ConfigHelper.get_active_config_or_default()
+    def legacy_generate_llm_chain(self, question: str, sources: list[Document]):
         answering_prompt = PromptTemplate(
-            template=config.prompts.answering_prompt,
+            template=self.config.prompts.answering_prompt,
             input_variables=["question", "sources"],
         )
 
-        llm_helper = LLMHelper()
+        sources_text = "\n\n".join(
+            [f"[doc{i+1}]: {source.page_content}" for i, source in enumerate(sources)]
+        )
+
+        return answering_prompt, {
+            "question": question,
+            "sources": sources_text,
+        }
+
+    def generate_llm_chain(
+        self,
+        question: str,
+        chat_history: list[dict],
+        sources: list[Document],
+    ):
+        examples = []
+
+        few_shot_example = {
+            "documents": self.config.example.documents.strip(),
+            "user_question": self.config.example.user_question.strip(),
+            "answer": self.config.example.answer.strip(),
+        }
+
+        # Remove whitespace
+        if few_shot_example["documents"]:
+            few_shot_example["documents"] = json.dumps(
+                json.loads(few_shot_example["documents"])
+            )
```

Comment:
To avoid the need for a comment, should we extract this to a function?

Reply:
Done, and updated the tests as I realised they weren't quite correct